src/indexing/enclave_lfs_access.h

```diff
@@ -0,0 +1,282 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "ccf/indexing/lfs_interface.h"
+#include "crypto/entropy.h"
+#include "crypto/hash.h"
+#include "crypto/symmetric_key.h"
+#include "ds/hex.h"
+#include "ds/messaging.h"
+
+#include <optional>
+#include <set>
+#include <unordered_map>
+
+// Uncomment to disable encryption and obfuscation, writing cache content
+// directly unencrypted to host disk
+// #define PLAINTEXT_CACHE
+
+namespace ccf::indexing
+{
+  static inline std::vector<uint8_t> get_iv(const LFSKey& key)
+  {
+    auto h = crypto::SHA256((const uint8_t*)key.data(), key.size());
+    h.resize(crypto::GCM_SIZE_IV);
+    return h;
```

Comment:
If the bucket `key` is re-used to encrypt multiple content (which I believe it is?), then the AES GCM IV will be re-used, compromising the encryption key. I think it'd be better to use a monotonic counter for the IV instead.

Reply:
So the same key shouldn't be reused for different content (it'll only be reused to rewrite the same bucket, which should contain the same contents). But I keep thinking the IV needs to be independently derivable and of course it doesn't - stamping a monotonic counter makes more sense, I'll do that.

Reply:
I said monotonic, but actually implemented random. This introduced an issue caught by one of the unit tests, where a malicious host could substitute `file_b` for `file_a` - both are encrypted and self-consistent, we just have completely the wrong contents. To avoid _that_, I've prefixed the plaintext with the file name before encrypting, so we still have a cheap check for "is this what I asked for" on correctly decrypted contents. We could also do that by keeping part of the IV for a hash of the filename, or using per-file HKDF derived keys (I'm leaning towards the latter, just need to work out the details).